### DISCUSSION TITLE
Is cutlass hopper warp specialization gemm requires M alignment?

### DISCUSSION THREAD

**Discussion by MARD1NO** [NVIDIA/cutlass#1500] [Q&A] [answered]
Recently I want to run M=2 K=8192 N=7168 54_hopper_fp8_warp_specialized_gemm, it raise a cutlass error:
```
Got cutlass error: Invalid status at: 488
```

but in M=16/32 it works well, is there exist some configurations I need to change to support M=2 gemm？

**Comment by thakkarV** [comment, marked as answer]
Is your layout M major? If so, it has to respect TMA alignment.

**Reply by MARD1NO** [reply]
I just use example's layout, A is RowMajor, B is ColMajor, C is ColMajor（I think it can get the best performance since fp8 tensorcore requires this layout?), and I'm not very clear about what you said `M Major`
<img width="655" alt="image" src="https://github.com/NVIDIA/cutlass/assets/42901638/4d2cccea-a2cb-43c7-9d0c-deaad09131cb">

**Reply by MARD1NO** [reply]
emmm I change C -> RowMajor it works
